libs/ledger-live-common/src/market/v2/useMarketDataProvider.ts

```
@@ -0,0 +1,128 @@
+
+  const liveCoinsList = useMemo(
+    () =>
+      supportedCurrencies?.filter(({ id }) => liveCompatibleIds.includes(id)).map(({ id }) => id),
+    [liveCompatibleIds, supportedCurrencies],
+  );
+
+  return {
+    supportedCounterCurrencies,
+    supportedCurrencies,
+    liveCoinsList,
+  };
+}
+
+export function useCurrencyChartData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyChartData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyChartData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  return results;
+}
+
+export function useCurrencyData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  const resultCurrency = useQuery({
+    queryKey: [QUERY_KEY.CurrencyDataRaw, id],
+    queryFn: () => fetchCurrency({ id }),
+    refetchInterval: REFETCH_TIME,
+    staleTime: REFETCH_TIME,
+  });
+
+  return { currencyDataByRanges: results, currencyInfo: resultCurrency };
+}
+
+export function useSupportedCounterCurrencies() {
+  const { isPending, error, data, isFetching } = useQuery({
+    queryKey: [QUERY_KEY.SupportedCounterCurrencies],
+    queryFn: () => supportedCounterCurrencies(),
+  });
+
+  return {
+    isPending,
+    error,
+    data,
+    isFetching,
+  };
+}
+
+export function useSupportedCurrencies() {
+  const { isPending, error, data, isFetching } = useQuery({
+    queryKey: [QUERY_KEY.SupportedCurrencies],
+    queryFn: () => getSupportedCoinsList(),
+  });
+
+  return {
+    isPending,
+    error,
```

where are we capturing the error, if there is one, could be a good idea to use something like ... `captureException from @sentry/nextjs` possibly in the getSupportedCoinsList() or earlier to fail fast, capture context sooner etc.